enumfields/fields.py

```diff
@@ -34,7 +36,11 @@ def to_python(self, value):
         raise ValidationError('%s is not a valid value for enum %s' % (value, self.enum), code="invalid_enum_value")
 
     def get_prep_value(self, value):
-        return None if value is None else self.enum(value).value
+        if value is None:
+            return None
+        if not isinstance(value, self.enum):
+            return self.enum(value).value
+        return value.value  # Already the correct type
```

Comment:
Performance testing a single function out of context doesn't tell much, since like in this case, if it's faster for some cases and slower for some others, then it's hard to judge which version is be better overall. Anyway, as I said, the longer version is more explicit and therefore easier to understand and that's enough for me.